moshi/src/main/java/com/squareup/moshi/JsonWriter.kt

```diff
@@ -0,0 +1,555 @@
+/*
+ * Copyright (C) 2010 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.squareup.moshi
+
+import com.squareup.moshi.JsonScope.EMPTY_ARRAY
+import com.squareup.moshi.JsonScope.EMPTY_OBJECT
+import com.squareup.moshi.JsonScope.NONEMPTY_ARRAY
+import com.squareup.moshi.JsonScope.NONEMPTY_OBJECT
+import okio.BufferedSink
+import okio.BufferedSource
+import java.io.Closeable
+import java.io.Flushable
+import java.io.IOException
```

Comment:
Please use the `kotlin.io` or `okio` versions of these

Reply:
There's not `Flushable` found, what's the replacement?

Reply:
ah ok, leave that one for now

Reply:
8e7fe11ac140a77b06fea18b91272cf41a01ccec